runtime/gc_modron_startup/mminit.cpp

```diff
@@ -2719,6 +2719,17 @@ gcInitializeDefaults(J9JavaVM* vm)
 		goto error;
 	}
 
+	if ((-1 == memoryParameterTable[opt_Xms]) && (-1 != memoryParameterTable[opt_initialRAMPercent])) {
+		extensions->initialMemorySize = (uintptr_t)(((double)extensions->usablePhysicalMemory / 100.0) * extensions->initialRAMPercent);
+		/* Update memory parameter table to appear that -Xms was specified */
+		memoryParameterTable[opt_Xms] = memoryParameterTable[opt_initialRAMPercent];
```

Comment:
This means that in case of error related to out-of-range case (<1m for example) output message would mention -Xms option. This fact should be documented at least.

Reply:
How about updating the message to say " -Xmx (as set by -XX:InitialRAMPercentage)"  when -XX:MaxRAMPercentage is being used to set max heap size. Some examples of updated messages:
* Option too large: -Xmx (as set by -XX:InitialRAMPercentage) is too large
* -Xmx (as set by -XX:InitialRAMPercentage) too small, must be at least 1 Mbytes

I can add a function that returns "-Xmx" or "-Xmx (as set by -XX:InitialRAMPercentage) " based on the option being used:

```
const char *
displayXmsOrInitialRAMPercentage(IDATA* memoryParameters)
{
        if ((-1 == memoryParameters[opt_initialRAMPercent])
                && (memoryParameters[opt_Xms] == memoryParameters[opt_initialRAMPercent])
        ) {
                return "-Xms (as set by -XX:InitialRAMPercentage)";
        } else {
                return "-Xms";
        }
}
```
and we can use this function wherever we are currently displaying "-Xmx".
Does that look fine?

Reply:
yes, something like this